clean up a couple typos and unclear parts in TypeScript workflow docs (#814)
temporalio/documentation: docs/typescript/workflows.md

If you wanted to send data in, you probably will want to read data out.
Since both involve communicating with a Workflow, using them is a two step process:

> 1. add the signals and queries inside the Workflow code
> 2. call the signals and queries from the Client code

Reviewer: ```suggestion
1. add the Signals and Queries inside the Workflow code
2. call the Signals and Queries from the Client code
```

- In other words, they do not "lock" the process, allowing one Worker to concurrently process hundreds of Workflows that sleep and await arbitrary conditions.

> - They are also "cancellation aware", allowing for graceful cleanup if the Workflow they are linked to is canceled. More in [Cancellation Scopes](/docs/typescript/cancellation-scopes).

Reviewer 2: ```suggestion
- They are also "cancellation aware", allowing for graceful cleanup if the Workflow they are linked to is canceled.
 For more information, see [Cancellation Scopes](/docs/typescript/cancellation-scopes).
```

> The Workflow's v8 isolate environment completely replaces the JavaScript [`setTimeout`](https://typescript.temporal.io/api/namespaces/workflow/#timers) global, including inside libraries that you use, to provide a complete JS runtime.

Reviewer 2: ```suggestion
The Workflow's V8 isolate environment completely replaces the JavaScript [`setTimeout`](https://typescript.temporal.io/api/namespaces/workflow/#timers) global, including inside libraries that you use, to provide a complete JavaScript runtime.
```

> We recommend using our [`sleep(timeout)`](https://typescript.temporal.io/api/namespaces/workflow/#sleep) API instead, as it is a cancellation-aware Promise wrapper for `setTimeout`.

Reviewer 2: ```suggestion
We recommend using our [`sleep(timeout)`](https://typescript.temporal.io/api/namespaces/workflow/#sleep) API instead, because it is a cancellation-aware Promise wrapper for `setTimeout`.
```
